### PR TITLE
fixed env updater and added error for missing .env

### PR DESCRIPTION
**Pull request by maximishchuk**
- moved check_env code to be called in the rundev command
- made it so that the env template updater doesn't read/write blank lines
- script now raises an error and stops the program when a .env file isn't detected

### CODE REVIEW

**Review by schromya — CHANGES_REQUESTED** [review]
+ Still need to raise error instead of `print(f".env file missing variable {check_line}")`
+ Convert everything to camel case (it's our style standard) and run `black app api` to format your code. Run `pylint app/ api/` and see if there's any issues too.
+ Blank lines are still being added to the env_check_file (not really that big of a deal if it's too hard to fix).

File: app/management/commands/rundev.py [hunk 1]
```diff
@@ -2,6 +2,63 @@
...
+        exit()
+    try:
+        env_check_file = open(os.path.join(file_dir,'env_check_file'))
+    except FileNotFoundError:
+        print('env_check_file not found, creating...')
+        env_check_file = open(os.path.join(file_dir,'env_check_file'), 'x')
+        env_check_file = open(os.path.join(file_dir,'env_check_file'))
+    counter = 0
+    for check_line in env_check_file:
+        counter += 1
```
Comment (crazykyler005) [inline app/management/commands/rundev.py]: counter is not being used

File: app/management/commands/rundev.py [hunk 2]
```diff
@@ -2,6 +2,63 @@
...
+        check_line = check_line.replace("\n","")
+        check_line = check_line.replace(" ","")
+        if check_line == "":
+            continue
+        env_file_lines = open(os.path.join(file_dir,'.env'))
+        for env_line in env_file_lines:
+            if env_line.replace(" ","").replace("\n","") == "":
+                continue
+            if check_line in env_line.partition("=")[0]:
+                LINE_EXISTS = True
```
Comment (crazykyler005) [inline app/management/commands/rundev.py]: there is a lot of cleanup that can happen here:
```suggestion
file_dir = os.path.dirname(os.path.realpath('__file__'))
env_file_path = os.path.join(file_dir, ".env")
env_checker_path = os.path.join(file_dir, "env_check_file")

def check_env():
    # checks .env for all the required fields
    # print('Checking Environment Variables...')
    
    if not os.path.exists(env_file_path):
        print("The .env file does not exist, you should create it!")
        exit()
    	
    if not os.path.exists(env_checker_path):
        os.mknod(env_checker_path)
        print('env_check_file not found, creating...')
        
    checker_properties = []
    
	# using "with open" ensures that the file gets opened and closed without having to explicitly call f.close()
    with open(env_checker_path, "r") as f:
        checker_properties = f.read().splitlines()
    
    for property in checker_properties:
        property.replace(" ","").replace("\r",")
        LINE_EXISTS = False
        
		# if is empty
        if not property:
            continue

        env_properties = []
        with open(env_file_path, "r") as f:
            env_properties = f.readlines()
            
        for env_line in env_properties:
            if env_line.find(property) != -1:
                LINE_EXISTS = True
                break
            
        if LINE_EXISTS == False:
            print(f".env file missing variable {property}")
```

File: app/management/commands/rundev.py [hunk 3]
```diff
@@ -16,6 +73,9 @@ def handle(self, *args, **options):
             shell=True,
         )
 
+        check_env()
```
Comment (crazykyler005) [inline app/management/commands/rundev.py]: should we have this return a true or false value and auto-quit if we are missing a known property?

File: app/management/commands/rundev.py [hunk 4]
```diff
@@ -2,6 +2,63 @@
...
+                break
+        if LINE_EXISTS == False:
+            print(f".env file missing variable {check_line}")
+
+def check_env_updater():
+    # automatically updates env_check_file with all the variables from .env
+    # print('Updating env_check_file...')
+    file_dir = os.path.dirname(os.path.realpath('__file__'))
+    env_file = open(os.path.join(file_dir,'.env'))
+    env_check_file = open(os.path.join(file_dir,'env_check_file'))
```
Comment (crazykyler005) [inline app/management/commands/rundev.py]: if your opening a file like this make sure you close it or use `with open` to automatically do so

File: app/management/commands/rundev.py (hunk 1 above)
Comment (crazykyler005) [reply app/management/commands/rundev.py]: also, for future reference, a cleaner way to increment count is to use `enumerate` with your for statement:
```
	for count, check_line in enumerate(env_check_file):
```

File: app/management/commands/rundev.py (hunk 2 above)
Comment (crazykyler005) [reply app/management/commands/rundev.py]: same for the checker_env_update function:
```
def check_env_updater():
    env_properties = []
    with open(env_file_path,'.r') as f:
        for line in f.read().splitlines():
            env_properties.append(line.split("=")[0].replace(" ",""))
```

**Review by schromya — CHANGES_REQUESTED** [review]
I'm getting the message `.env file missing manditory MONGO_USERNAME` although MONGO_USERNAME is in my .env

**Comment by crazykyler005** [comment]
> I'm getting the message `.env file missing manditory MONGO_USERNAME` although MONGO_USERNAME is in my .env

I'm not able to replicate this issue....

**Review by schromya — APPROVED** [review]
Bug is fixed :D

**Pull request merged**